prow/apis/prowjobs/v1/types.go

```diff
@@ -344,10 +344,13 @@ type DecorationConfig struct {
 	GCSConfiguration *GCSConfiguration `json:"gcs_configuration,omitempty"`
 	// GCSCredentialsSecret is the name of the Kubernetes secret
 	// that holds GCS push credentials.
-	GCSCredentialsSecret string `json:"gcs_credentials_secret,omitempty"`
+	GCSCredentialsSecret *string `json:"gcs_credentials_secret,omitempty"`
 	// S3CredentialsSecret is the name of the Kubernetes secret
 	// that holds blob storage push credentials.
 	S3CredentialsSecret string `json:"s3_credentials_secret,omitempty"`
```

Comment:
nit, make this a pointer to make things consistent

Reply:
Good call, it's needed for more than just consistency. If someone wants to override a broadly scoped decoration config using S3 credentials in favor of a K8s service account with workload identity they'll need this in order to unset the S3 credentials.
Fixed.